### PR TITLE
feat(repl): Add color to functions

### PR DESCRIPTION
**Pull request by sigmaSd**
This pr adds colors to functions
image

This is all subjective but I thought its nicer to have more colors in general.

### CODE REVIEW

File: cli/tools/repl/editor.rs [hunk 1]
```diff
@@ -304,7 +305,9 @@ impl Highlighter for EditorHelper {
   fn highlight<'l>(&self, line: &'l str, _: usize) -> Cow<'l, str> {
     let mut out_line = String::from(line);
 
-    for item in deno_ast::lex(line, deno_ast::MediaType::TypeScript) {
+    let mut lexed_items: VecDeque<_> =
+      deno_ast::lex(line, deno_ast::MediaType::TypeScript).into();
```
Comment (dsherret) [inline cli/tools/repl/editor.rs:309]: Maybe using `peakable()` might be more appropriate (https://doc.rust-lang.org/std/iter/struct.Peekable.html)? Then you can use `lexed_items.next()` and `lexed_items.peek()`?

File: cli/tools/repl/editor.rs [hunk 2]
```diff
@@ -334,7 +337,14 @@ impl Highlighter for EditorHelper {
                 } else if ident == *"async" || ident == *"of" {
                   colors::cyan(&line[range]).to_string()
                 } else {
-                  line[range].to_string()
+                  let next = lexed_items.get(0).map(|item|&item.inner);
+                  if matches!(next, Some(deno_ast::TokenOrComment::Token(next_token)) if matches!(next_token, Token::LParen)) {
+                    // We're looking for something that looks like a function
+                    // We use a simple heuristic: 'ident' followed by 'LParen'
+                    colors::intense_blue(&line[range]).to_string()
```
Comment (dsherret) [inline cli/tools/repl/editor.rs:344]: I like it. Is there any precedence where other people are doing this too?

File: cli/tools/repl/editor.rs (hunk 1 above)
Comment (sigmaSd) [reply cli/tools/repl/editor.rs]: done

File: cli/tools/repl/editor.rs (hunk 2 above)
Comment (sigmaSd) [reply cli/tools/repl/editor.rs:344]: Not sure, I just used this trick before for a repl I have and it worked well

Comment (sigmaSd) [reply cli/tools/repl/editor.rs:344]: But its just a heuristic, so it only works for the *most* part, which I think its fine personally for a repl.

Comment (bartlomieju) [reply cli/tools/repl/editor.rs:344]: Maybe we should apply same coloring as in diagnostics coming from TSC or the linter?

Comment (dsherret) [reply cli/tools/repl/editor.rs:344]: Oh, by precedence I just meant someone else using this colour here. The heuristic looks good. Anyway, not a big deal.

File: cli/tools/repl/editor.rs [hunk 3]
```diff
@@ -334,7 +337,14 @@ impl Highlighter for EditorHelper {
                 } else if ident == *"async" || ident == *"of" {
                   colors::cyan(&line[range]).to_string()
                 } else {
-                  line[range].to_string()
+                  let next = lexed_items.peek().map(|item|&item.inner);
+                  if matches!(next, Some(deno_ast::TokenOrComment::Token(next_token)) if matches!(next_token, Token::LParen)) {
```
Comment (dsherret) [inline cli/tools/repl/editor.rs:341]: This doesn't work?

```
matches!(next, Some(deno_ast::TokenOrComment::Token(Token::LParen))))
```

**Review by dsherret — APPROVED** [review]
LGTM

File: cli/tools/repl/editor.rs (hunk 3 above)
Comment (sigmaSd) [reply cli/tools/repl/editor.rs]: yep works

**Pull request merged**